src/runtime/particle.ts

```diff
@@ -56,13 +56,19 @@ export class Particle {
    * Called after handles are synced, override to provide initial processing.
    */
   protected ready(): void {
+    this.onReady();
   }
 
   /**
    * Called after handles are writable, only on first initialization of particle.
    */
   onCreate(): void {}
 
+  /**
```

Comment:
Do we have any guarantees that onCreate() will always happen before onReady? I hope we do, and that we document that here.

Reply:
In the Kotlin ArcHost, this is guaranteed, onCreate() is always called first immediately after instantiation and handle creation. (only once)

Reply:
If we don't make that commitment on all platforms, a particle author can't count on being able to do all initialization in onCreate().

Reply:
We are. Ray has been talking about ArcHost as that's closest to his heart right now, but this guarantees are given across all platforms.

Reply:
Thanks for bringing this up, it got me thinking more about the order and I think it is all ensured properly now. I also added more tests to make certain things work as intended.